kratos/utilities/sparse_matrix_multiplication_utility.h

```diff
@@ -683,6 +683,267 @@ class SparseMatrixMultiplicationUtility
         }
     }
 
+    /**
+     * @brief This method assembles several sparse matrices into one large sparse matrix
+     * @param rMatricespBlocks The pointers to the matrices we are interested in assemble
+     * @param ContributionCoefficients The matrix containing the coefficients to be considered (copy, so we don't need to provide it)
+     * @param TransposeBlocks The matrix containing the flags telling us to transpose the blocks (copy, so we don't need to provide it)
+     */
+    template <class TMatrix>
+    static inline void AssembleSparseMatrixByBlocks(
+        TMatrix& rMatrix,
+        const DenseMatrix<TMatrix*>& rMatricespBlocks,
+        DenseMatrix<double> ContributionCoefficients = DenseMatrix<double>(0,0),
+        DenseMatrix<bool> TransposeBlocks = DenseMatrix<bool>(0,0)
+        )
+    {
+        const SizeType number_of_rows_blocks = rMatricespBlocks.size1();
+        const SizeType number_of_columns_blocks = rMatricespBlocks.size2();
+
+        // Fill the matrices if they are empty
+        if (ContributionCoefficients.size1() == 0 && ContributionCoefficients.size2() == 0) {
```

Comment:
what is the expected size of "ContrubtionCoefficients"?

Reply:
The number of blocks in each direction